.github/workflows/msvc-full-features.yml

```diff
@@ -79,17 +92,74 @@ jobs:
       run: |
         vcpkg integrate install
 
+    - name: Download ccache
+      uses: robinraju/release-downloader@v1.4
+      with:
+        repository: 'ccache/ccache'
+        tag: 'v4.6.1'
+        fileName: 'ccache-4.6.1-windows-x86_64.zip'
+
+    - name: Install ccache
+      id: install-ccache
+      run: |
+        unzip ccache-4.6.1-windows-x86_64.zip
+        cp ccache-4.6.1-windows-x86_64/ccache.exe ccache-4.6.1-windows-x86_64/cl.exe
+        cp ccache-4.6.1-windows-x86_64/ccache.exe ccache-4.6.1-windows-x86_64/clang-cl.exe
+        mv ccache-4.6.1-windows-x86_64 ${{ env.CDDA_CCACHE_PATH }}
+
+    - name: Get ccache vars
+      id: get-vars
+      run: |
+        echo "::set-output name=datetime::$(/bin/date -u "+%Y%m%d%H%M")"
+        echo "::set-output name=ccache-path::$(echo "$APPDATA\\ccache")"
+      shell: bash
+
+    - name: ccache cache files
+      uses: actions/cache@v3
+      with:
+        path: ${{ steps.get-vars.outputs.ccache-path }}
+        # double-dash after compiler is not a typo, it is to disambiguate between g++-<date> and g++-11-<date> for restore key prefix matching
+        key: ccache-${{ github.ref_name }}-${{ runner.os }}-msvc--${{ steps.get-vars.outputs.datetime }}
+        restore-keys: |
+          ccache-master-${{ runner.os }}-msvc--
```

Comment:
This is all based off the matrix builds ccache configuration, so its a known pattern.

https://github.com/CleverRaven/Cataclysm-DDA/blob/master/.github/workflows/matrix.yml#L260-L275